bounded-collections/src/bounded_btree_map.rs

```diff
@@ -29,6 +29,7 @@ use core::{borrow::Borrow, marker::PhantomData, ops::Deref};
 ///
 /// Unlike a standard `BTreeMap`, there is an enforced upper limit to the number of items in the
 /// map. All internal operations ensure this bound is respected.
+#[cfg_attr(feature = "std", derive(Hash))]
```

Comment:
```suggestion
#[cfg_attr(feature = "std", derive(Hash, serde::Serialize, serde::Deserialize), serde(transparent))]
```

Reply:
I'm not sure about deriving these; `Deserialize` would need a custom `impl` to add bounds checks, like we do now with `BoundedVec`. And the custom `impl` looks annoying to implement, so I'm thinking we don't add it until there's actually a need for it.

Reply:
@bkchr I'm going to go ahead and merge -- these derives can be a follow-up if we really want them.